docs - add information on upgrading to PostGIS 2.5.4

mkiyama opened the pull request:
Upgrade instructions 2.1.5 to different versions of 2.5.4
Includes upgrade to PostGIS 2.5.4.pivotal.3 package.

Will be backported to 6X_STABLE only.

Link to HTML output on a temporary GPDB doc site.
https://docs-msk-gpdb7-dev.cfapps.io/7-0/analytics/postgis-upgrade.html

fmcquillan99 commented:
I got very confused reading this document, I am afraid our users will also be very confused when they want to upgrade.  Below are some comments but maybe a meeting with Mel/Ekta/me might help.

(1)
When you say:
``` 
Important: Before removing PostGIS support from a database, ensure that no users are accessing the database. Removing PostGIS support from a database drops PostGIS database objects from the database. Users accessing PostGIS objects might interfere with the dropping of PostGIS objects. 
```
this may be confusing to the user in its current form.  This page is about `Upgrading from PostGIS 2.1.5 or 2.5.4` and this note is about removing PostGIS, which is not required for the upgrade process.  Removing an old PostGIS and re-installing a new PostGIS is the destructive path we are replacing with the new upgrade script.

But having said that, I think there ought to be a clear warning at the top of the page like you have it about the dangers of dropping the extension.  Maybe just some rewording, like:

```
For Greenplum Database 6, you can upgrade from PostGIS 2.1.5 to 2.5.4, or from a PostGIS 2.5.4 package to a newer PostGIS 2.5.4 package using the upgrade tool described on this page.
```
page TOC
``` 
Important: Upgrading PostGIS using the upgrade tool does not require you to remove PostGIS and re-install it.  In fact, removing PostGIS from a database drops PostGIS database objects from the database, potentially without warning.  etc about active user sessions and such
```


(2)
```
Upgrading PostGIS 2.5.4 package from pivotal.1 or pivotal.2 to pivotal.3
Upgrades the PostGIS pivotal.2 package to the pivotal.3 minor release package.
```
Should read:
```
Upgrading PostGIS 2.5.4 package from pivotal.1 or pivotal.2 to pivotal.3
Upgrades the PostGIS pivotal.1 or pivotal.2 package to the pivotal.3 minor release package.
```

(3)
```
Upgrading the postgis-2.5.4 Package from pivotal.1 to pivotal.2
Upgrades the PostGIS package to the pivotal.2 minor release package. The minor release supports using the CREATE EXTENSION command and the DROP EXTENSION command to enable and remove PostGIS support in a database.
```
I think we should remove this altogether ^^^ since there is no reason why a user would upgrade to pivotal.2.  They would upgrade to pivotal.3 .  Unless I am missing something ???


(4)
```
Upgrade the postgis-2.5.4+pivotal.2 package from build.1 to build.2
Upgrades the PostGIS package to the build.2 maintenance release. The release is an ABI (Application Binary Interface) compatible PostGIS package that contains the same PostGIS version (2.5.4).
```
Will be the similar for `postgis-2.5.4+pivotal.3` . We should be more general about perhaps about describing ABI compatability.


(5)
For the whole section
```
Upgrading the postgis-2.5.4 Package from pivotal.1 to pivotal.2
```
again, I am not sure why we need this, unless I am missing something ???


(6)
```
Upgrade the postgis-2.5.4+pivotal.2 package from build.1 to build.2
```
I assume you are using `postgis-2.5.4+pivotal.2` as an example?  This could apply to any other dot version too, right?  Perhaps we should state that.


(7)
The links from the TOC at the top seem to jump to the wrong place on the page:
```
Upgrade a PostGIS 2.5.4 Package
Upgrade from PostGIS 2.1.5 to PostGIS 2.5.4
Checking the PostGIS Version
```

(8)
```
Important: Upgrading to PostGIS 2.5.4 requires removing PostGIS 2.1.5 support from databases with PostGIS enabled. Removing PostGIS support drops the PostGIS 2.1.5 objects from the database. Before removing PostGIS 2.1.5 support, ensure that no users are accessing the database. Users accessing PostGIS objects might interfere with dropping PostGIS objects.
```
This was for 2.1.5 to 2.5.4+pivotal.1/ 2.5.4+pivotal.2 but it is not the case for 2.5.4+pivotal.3+.
We don't want anyone to upgrade to from 2.1.5 to 2.5.4+pivotal.1/ 2.5.4+pivotal.2


(9)
```
Notes
Upgrading from a PostGIS 2.1.5 package to a PostGIS 2.5.4 package requires removing support for PostGIS 2.1.5 from a database with the postgis_manager.sh script. The script drops the PostGIS objects from the database that were created with PostGIS 2.1.5.
```
Again, not the case for moving to 2.5.4+pivotal.3+

mkiyama commented:
@fmcquillan99 @khannaekta 

Updated the upgrade PostGIS doc.

Removed a lot of upgrade options. Now only document
-2.1.5 to 2.5.4 pivotal.3
-2.5.4 (pivotal.1 and pivotal.2) to 2.5.4 pivotal.3
- moved note about removing PostGIS support from a database to the Removing PostGIS section.

fmcquillan99 commented:
Looks much clearer, thanks for the edits.  Couple more comments:

100)
`After upgrading the PostGIS package, you can remove the PostGIS 2.1.5 package.`

This may be confusing. I think you are talking about the `gppkg` when you say "package", but the user may think that the old 2.1.5 extension gets left behind and they need to manually remove it?  - which I don't think is necessary, right Ekta?  Suggest we make the clear distinction between cleaning up `gppkg` files and the extension itself.

101)
`Starting with the postgis-2.5.4+pivotal.2 package, you enable support PostGIS in a database with the CREATE EXTENSION command.`
* fix wording

khannaekta commented:
@fmcquillan99 When upgrading from 2.5.4 --> 2.1.5, since we install(`gppkg -i`) the gppkg for 2.5.4 (instead of doing an upgrade`gppkg -u`), the gppkg for 2.1.5 is also installed, so the user will have to explicitly remove it. This is just being more conservative because if we directly upgrade to 2.5.4 the gppkg for 2.1.5 will get removed and if anything fails when running `postgis_manager.sh` script with the upgrade option, it may lead PostGIS in an inconsistent state.

fmcquillan99 commented:
Oh I see, in that case ignore my comment (100).

khannaekta commented on gpdb-doc/dita/analytics/postgis-upgrade.xml:
```diff
@@ -0,0 +1,136 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
+<topic id="topic_khf_ltc_vbb">
+  <title>Upgrading from PostGIS 2.1.5 or 2.5.4</title>
```
we should probably update the title to Just `Upgrading PostGIS` as it has both from 2.1.5-> 2.5.4 as well as 2.5.4 to a newer 2.5.4.

fmcquillan99 commented:
@mkiyama after final edits could you please publish a final WIP HTML file so we can have a last look?

mkiyama commented:
@fmcquillan99 @khannaekta 

The docs and HTML site have been updated.

Thanks for all your help.
